### ISSUE TITLE
Json property order in System.Text.Json

### ISSUE THREAD

**Issue by kotx** [dotnet/runtime#43415]
Users should be able to specify the order of a json property during serialization. 
It is possible to specify order by changing the order of the properties in the code. However, it is not possible to do this in inherited classes. 

#### Example
```cs
// FooView.cs
public class FooView : BaseView
{
    public string Foo { get; set; }
}
```
```cs
// BaseView.cs
public class BaseView
{
    public bool Success { get; set; }
}
```
`FooView`'s properties will always be placed before `BaseView`'s upon serialization.
```json
{
    "Foo": "bar",
    "Success": true
}
```

#### Implementation
Newtonsoft.Json does it like this:
```cs
[JsonProperty(Order = -2)]
public string Foo { get; set; }
```
~~An idea for System.Text.Json could be to use the `DataMemberAttribute` instead of creating its own attribute.~~

**Comment by am11** [comment]
Duplicate of #728.

**Comment by kotx** [comment]
Closed in favor of #728.
